Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/UserManager.java

```diff
@@ -317,8 +314,13 @@ public boolean addNewUser(AccountRequest request) throws QuadrigaStorageExceptio
             throw new UsernameExistsException("Username already in use.");
 
         String plainPassword = request.getPassword();
-        boolean success = usermanagerDAO.addNewUserAccountRequest(request.getUsername(), encryptPassword(plainPassword),
-                request.getName(), request.getEmail());
+        boolean success = false;
+        
+        if(request.isSocialSignIn()){
+            success =  usermanagerDAO.addNewUserAccountRequest(request.getUsername(),  null, request.getName(),  request.getEmail(),  request.getProvider(),  request.getUserIdOfProvider());
+        }else{
+            success = usermanagerDAO.addNewUserAccountRequest(request.getUsername(), encryptPassword(plainPassword), request.getName(), request.getEmail(), null, null);
+        }
```

Comment:
Wouldn't ```request.getProvider()``` and ```request.getUserIdOfProvider()``` be null anyways if the request is not a social sign in? And ```encryptPassword(plainPassword)``` can be simply executed with a tertiary operator in line 316. Then you can avoid the whole if statement.